src/coreComponents/fieldSpecification/FieldSpecificationManager.cpp

```diff
@@ -193,20 +193,40 @@ void FieldSpecificationManager::validateBoundaryConditions( MeshLevel & mesh ) c
 
     if( isFieldNameFound == 0 )
     {
-      char const fieldNameNotFoundMessage[] =
-        "\n{}: there is no {} named `{}` under the {} `{}`.\n";
-      string const errorMsg =
-        GEOS_FMT( fieldNameNotFoundMessage,
+      std::ostringstream fieldNameNotFoundMessage;
+      std::string fieldNamePath =
+        GEOS_FMT( "\n{}: there is no {} named `{}` under the {} `{}`.\n",
                   fs.getWrapperDataContext( FieldSpecificationBase::viewKeyStruct::fieldNameString() ),
                   FieldSpecificationBase::viewKeyStruct::fieldNameString(),
-                  fs.getFieldName(), FieldSpecificationBase::viewKeyStruct::objectPathString(), fs.getObjectPath() );
+                  fs.getFieldName(), FieldSpecificationBase::viewKeyStruct::objectPathString(), fs.getObjectPath());
+
+      fieldNameNotFoundMessage << fieldNamePath;
       if( areAllSetsEmpty )
       {
-        GEOS_LOG_RANK_0( errorMsg );
+        GEOS_LOG_RANK_0( fieldNameNotFoundMessage.str() );
       }
       else
       {
-        GEOS_THROW( errorMsg, InputError );
+        fieldNameNotFoundMessage << GEOS_FMT( "Available fields in {} are:\n", fs.getObjectPath());
+        bool stopIteration = false;
+        ElementRegionManager const & elemRegionGroup = mesh.getElemManager();
+        elemRegionGroup.forElementRegions( [&]( ElementRegionBase const & elemRegion )
+        {
+          if( stopIteration )
+            return;
+          auto const & subCellRegion = elemRegion.getSubRegion< CellElementSubRegion >( 0 );
+          if( !subCellRegion.hasWrapper( fs.getFieldName() ) )
+          {
+            stopIteration = true;
+            for( auto const & wrapperIter : subCellRegion.wrappers() )
+            {
+              auto const & wrapper = wrapperIter.second;
+              fieldNameNotFoundMessage << wrapper->getName() <<  " ";
```

Comment:
I would add a comma here 
```suggestion
              fieldNameNotFoundMessage << wrapper->getName() <<  ", ";
```